rnacentral_pipeline/databases/tarbase/parser.py

```diff
@@ -0,0 +1,148 @@
+import os
+
+import polars as pl
+import psycopg2 as pg
+from psycopg2.extras import execute_batch
+
+from rnacentral_pipeline.databases.data.entry import Entry
+from rnacentral_pipeline.databases.data.related import (
+    RelatedCoordinate,
+    RelatedEvidence,
+    RelatedSequence,
+)
+from rnacentral_pipeline.databases.helpers import phylogeny as phy
+
+
+def build_entry(row):
+    primary_id = row["mirna_id"]
+    ## Use this accession form to match existing accessions
+    accession = f"TARBASE:{row['mirna_id']}"
+
+    database = "TARBASE"
+
+    regions = (
+        []
+    )  # The genome coords provided by tarbase are for the PROTEIN gene, not miRNA. Give empty region?
+    rna_type = "SO:0000276"  # miRNA - tarBase is all miRNA
+
+    ## This takes you to the specific search results for the interation on this row
+    url = f"https://dianalab.e-ce.uth.gr/tarbasev9/interactions?gene={row['gene_name']}&mirna={row['mirna_name']}"
+
+    ## Not clear how to set this here
+    seq_version = 1
+
+    related_sequences = []
+    for (
+        related_gene,
+        exp_meth,
+    ) in zip(row["gene_id"], row["experimental_method"]):
+
+        evidence = RelatedEvidence(methods=[exp_meth])
+
+        related_sequences.append(
+            RelatedSequence(
+                sequence_id=related_gene,
+                relationship="target_protein",
+                evidence=evidence,
+            )
+        )
+
+    ## Make this as minimal as possible, don't fill everything
+    ent = Entry(
+        primary_id=primary_id,
+        accession=accession,
+        ncbi_tax_id=row["taxid"],
+        database=database,
+        sequence=row["sequence"].replace("U", "T"),
+        regions=regions,
+        rna_type=rna_type,
+        url=url,
+        seq_version=seq_version,
+        optional_id=row["mirna_name"],
+        description=f"{row['species']} ({phy.common_name(row['taxid']) }) {row['mirna_name']}",
+        note_data={"url": url},
+        xref_data={},
+        related_sequences=related_sequences,
+    )
+
+    return ent
+
+
+def parse(filepath):
+    """
+    Parse the provided tsv file into entry objects we can put in the database
+
+    Some notes on the TSV:
+    - mirna_name and mirna_id are the miRNA identifiers we will hold and should look for
```

Comment:
So on this - our mirbase accessions use the `mirna_id` format (e.g. MIMAT0000355), but our tarbase ones use the `mirna_name` format (e.g. TARBASE:dme-miR-210-3p). This is what we have in the database now. The new tarbase parser uses the MIMAT ID style for the accessions, because I thought it was clearer, but I guess I should change it to use the name?